web/domains/case/forms.py

```diff
@@ -303,3 +316,73 @@ class ReassignOwnershipExport(ReassignOwnershipBaseForm):
     class Meta:
         model = ExportApplication
         fields = ReassignOwnershipBaseForm.Meta.fields
+
+
+class DownloadDFLCaseDocumentsForm(forms.Form):
+    link: ImportApplicationDownloadLink | None
+    # The same error message for every outcome
+    ERROR_MSG = (
+        "Licence data not found using the provided values."
+        " Please ensure all details are entered correctly and try again."
+    )
+
+    def __init__(self, *args: Any, code: str, **kwargs: Any) -> None:
+        super().__init__(*args, **kwargs)
+        self.code = code
+        self.link = None
+
+    email = forms.EmailField(help_text="Enter the associated constabulary email address.")
+    constabulary = forms.ModelChoiceField(
+        queryset=Constabulary.objects.filter(is_active=True),
+        label="Constabulary",
+        widget=ModelSelect2Widget(
+            attrs={
+                "data-minimum-input-length": 3,
+                "data-placeholder": "-- Select Constabulary",
+            },
+            search_fields=("name__icontains",),
+        ),
+    )
+    check_code = forms.CharField(max_length=8, help_text="Enter the check code found in the email")
+
+    def clean(self):
+        # Load and store the link record if valid, display error if not.
+        cleaned = super().clean()
+
+        check_code = cleaned.get("check_code")
+        email = cleaned.get("email")
+        constabulary = cleaned.get("constabulary")
+
+        if not all((check_code, email, constabulary)):
```

Comment:
personal preference is multiple ands is more intuitive and doesn't require another function call

```suggestion
        if not (check_code and email and constabulary):
```

Reply:
Agreed, it was four fields initially so thats why I probaby used `all`, but I removed it before submitting the PR to review.